worksheets.go

```diff
@@ -320,6 +333,33 @@ func (ws *Worksheet) Set(name string, value Value) error {
 		return fmt.Errorf("Set on slice field %s, use Append, or Del", name)
 	}
 
+	if field.constrainedBy != nil {
```

Comment:
We should work on this a bit, there are now two different write paths (two branches), whereas we'd want the `constrainedBy` to more of a pre-processing.